File: backend/internal/entity/question.go
package entity

import (
	"time"

	"github.com/google/uuid"
	pq "github.com/lib/pq"
)

type Question struct {
	QuestionID    uuid.UUID      `gorm:"primaryKey" json:"question_id"`
	TryoutID      uuid.UUID      `json:"tryout_id"`
	LocalID       int            `json:"local_id"`
	Type          string         `json:"type"`
	Text          string         `json:"text"`
	ImageUrl      string         `json:"image_url"`
	IsOptions     *bool          `json:"is_options"`
	Options       pq.StringArray `gorm:"type:text[]" json:"options"`
	CorrectAnswer string         `json:"correct_answer"`
	Points        int            `json:"points"`
	CreatedAt     time.Time      `json:"created_at"`
	UpdatedAt     time.Time      `json:"updated_at"`
}

func (Question) TableName() string {
	return "questions"
}

File: cms/src/types/question.ts
export interface QuestionProps {
  question_id: string;
  tryout_id: string;
  local_id?: number;
  is_options?: boolean;
  type: string;
  text: string;
  options: string[];
  image_url?: string;
  correct_answer: string;
  points?: string;
  created_at?: Date | string;
  updated_at?: Date | string;
}

export interface CreateQuestionRequest {
  tryout_id: string;
  local_id?: number;
  type: string;
  text: string;
  options: string[];
  image_url?: string;
  correct_answer: string;
  points?: string;
  is_options?: boolean;
}

export interface UpdateQuestionRequest {
  question_id: string;
  tryout_id: string;
  local_id?: number;
  type?: string;
  text?: string;
  options?: string[];
  image_url?: string;
  correct_answer?: string;
  points?: string;
}
